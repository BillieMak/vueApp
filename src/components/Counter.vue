<template>
  <h2>{{ customTitle }}</h2>
  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
  <button @click="increment">+1</button>
  <button @click="decrement">-1</button>
</template>

<script>
export default {
  name: "Counter",
  props: {
    title: {
      type: String,
      // required: true,
    },
    start: {
      type: Number,
      default: 4,
      validator(value) {
        return value > 100;
      },
    },
  },
  data() {
    return {
      counter: this.start,
    };
  },
  methods: {
    increment() {
      this.counter = this.counter + 1;
    },
    decrement() {
      this.counter--;
    }
  },
  computed: {
    squareCounter() {
      return this.counter * this.counter;
    },
    customTitle() {
      return this.title || "Counter Custom"
    }
  }
};
</script>

<style>
button {
  background-color: #4CAF50;
  border-radius: 8px;
  border: none;
  color: white;
  display: inline-block;
  font-size: 16px;
  margin-right: 5px;
  padding: 10px;
  text-align: center;
  text-decoration: none;

}

button:hover {
  background-color: #3e8e41;
  cursor: pointer;
}
</style>