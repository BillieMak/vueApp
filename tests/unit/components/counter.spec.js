import { shallowMount } from "@vue/test-utils";
import Counter from "@/components/Counter";

describe("Counter Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Counter);
  });

  // test('Debe hacer un match con el snapshot', () => {

  //     const wrapper = shallowMount(Counter)

  //     expect(wrapper.html()).toMatchSnapshot()
  // })

  test('h2 debe tener el valor por defecto "Counter Custom"', () => {
    expect(wrapper.find("h2").exists()).toBeTruthy();

    const h2Value = wrapper.find("h2").text();

    expect(h2Value).toBe("Counter Custom");
  });

  test("el valor por defecto debe ser 4 en el p ", () => {
    expect(wrapper.findAll("p").length).toBe(2);

    // const pValue = wrapper.findAll("p")[1].text();

    const pValue = wrapper.find('[data-testid="counter"]').text();

    expect(pValue).toBe("4");
  });

  test("debe incrementar  y decrementar el contador", async () => {
    const [increaseBtn, decreaseBtn] = wrapper.findAll("button");

    await increaseBtn.trigger("click");

    // expect(pValue).toBe('5')

    await decreaseBtn.trigger("click");
    await decreaseBtn.trigger("click");

    const pValue = wrapper.find('[data-testid="counter"]').text();

    expect(pValue).toBe("3");
  });

  test("debe establecer el valor por defecto", () => {
    const { start } = wrapper.props();

    const value = wrapper.find('[data-testid="counter"]').text();

    expect(Number(value)).toBe(start);
  });

  test("debe de mostrar la prop title", () => {
    const title = "Hola mundo";

    const wrapper = shallowMount(Counter, {
      props: {
        title,
      },
    });

    expect(wrapper.find("h2").text()).toBe(title);
  });
});
