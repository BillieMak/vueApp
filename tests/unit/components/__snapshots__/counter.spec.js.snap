// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe hacer un match con el snapshot 1`] = `
<h2>Counter Custom</h2>
<p>4 <sup>2</sup> = 16</p>
<button>+1</button>
<button>-1</button>
`;
